webauthn-rs-core/src/core.rs

```diff
@@ -892,6 +919,7 @@ impl WebauthnCore {
         policy: Option<UserVerificationPolicy>,
         extensions: Option<RequestAuthenticationExtensions>,
         allow_backup_eligible_upgrade: Option<bool>,
+        hints: Option<Vec<PublicKeyCredentialHints>>,
```

Comment:
I feel like with a _4th_ optional parameter, now is a good time to move this function to a builder pattern, like `ChallengeRegisterBuilder` does.

I'm not sure whether this is a public API, but `WebauthCore::generate_challenge_authenticator` could then become a wrapper for the new builder pattern, and marked as deprecated.

Reply:
Yeah, probably. I'll do that now too.